hook-common/src/webhook.rs

```diff
@@ -124,16 +124,13 @@ pub struct WebhookJobParameters {
     pub headers: collections::HashMap<String, String>,
     pub method: HttpMethod,
     pub url: String,
+}
 
-    // These should be set if the Webhook is associated with a plugin `composeWebhook` invocation.
+/// `JobMetadata` required for the `WebhookConsumer` to execute a webhook.
+/// These should be set if the Webhook is associated with a plugin `composeWebhook` invocation.
+#[derive(Deserialize, Serialize, Debug, PartialEq, Clone)]
+pub struct WebhookJobMetadata {
     pub team_id: Option<i32>,
     pub plugin_id: Option<i32>,
     pub plugin_config_id: Option<i32>,
```

Comment:
@bretthoerner I think these should be unsigned non-optional types: Given that this struct only applies to Webhook jobs, it seems like all of these should be required metadata parameters, which we can enforce with non-optional types. Unsigned because I don't think negative ids could be generated. Any thoughts?

Reply:
Agreed, we can switch to `u32` for all of them I think.

> and if eventually that happens we can have a separate struct + endpoint to receive them perhaps?

Yeah, agree.